contracts/Cross-chain/OmnichainGovernanceExecutor.sol

```diff
@@ -0,0 +1,280 @@
+        );
+
+        Proposal storage proposal = proposals[proposalId_];
+
+        for (uint256 i; i < proposal.targets.length; ++i) {
+            proposalTimelocks[uint8(proposal.proposalType)].executeTransaction(
+                proposal.targets[i],
+                proposal.values[i],
+                proposal.signatures[i],
+                proposal.calldatas[i],
+                proposal.eta
+            );
+        }
+        proposal.executed = true;
+        emit ProposalExecuted(proposalId_);
+    }
+
+    /**
+     * @notice Cancels a proposal only if sender is the guardian and proposal is not executed.
+     * @param proposalId_ Id of proposal that is to be cancelled.
+     * @custom:access Sender must be the guardian.
+     * @custom:event Emits ProposalCancelled with proposal id of the cancelled proposal.
+     */
+    function cancel(uint256 proposalId_) external {
+        require(queued[proposalId_], "OmnichainGovernanceExecutor::cancel: proposal not queued");
+        Proposal storage proposal = proposals[proposalId_];
+        require(!proposal.executed, "OmnichainGovernanceExecutor::cancel: cannot cancel executed proposal");
+        require(msg.sender == GUARDIAN, "OmnichainGovernanceExecutor::cancel: sender must be guardian");
+
+        proposal.cancelled = true;
+        for (uint256 i = 0; i < proposal.targets.length; i++) {
+            proposalTimelocks[proposal.proposalType].cancelTransaction(
+                proposal.targets[i],
+                proposal.values[i],
+                proposal.signatures[i],
+                proposal.calldatas[i],
+                proposal.eta
+            );
+        }
+
+        emit ProposalCancelled(proposalId_);
+    }
+
+    function _blockingLzReceive(
+        uint16 srcChainId_,
+        bytes memory srcAddress_,
+        uint64 nonce_,
+        bytes memory payload_
+    ) internal virtual override whenNotPaused {
+        uint256 gasToStoreAndEmit = 30000; // enough gas to ensure we can store the payload and emit the event
+
+        (bool success, bytes memory reason) = address(this).excessivelySafeCall(
+            gasleft() - gasToStoreAndEmit,
+            150,
+            abi.encodeCall(this.nonblockingLzReceive, (srcChainId_, srcAddress_, nonce_, payload_))
+        );
+        // try-catch all errors/exceptions
+        if (!success) {
+            bytes32 hashedPayload = keccak256(payload_);
+            failedMessages[srcChainId_][srcAddress_][nonce_] = hashedPayload;
+            emit ReceivePayloadFailed(srcChainId_, srcAddress_, nonce_, reason); // Retrieve payload from the src side tx if needed to clear
+        }
+    }
+
+    function _nonblockingLzReceive(
+        uint16 srcChainId_,
+        bytes memory,
+        uint64,
+        bytes memory payload_
+    ) internal virtual override {
+        (bytes memory payload, uint256 pId) = abi.decode(payload_, (bytes, uint256));
```

Comment:
The `pId` is encoded as `uint64` in the Sender. Shouldn't we decode it using the same type?

Reply:
https://github.com/VenusProtocol/governance-contracts/pull/59/commits/c0e7529cdede8208a18a8042063b79e8da3929f3

Reply:
We will be using `uint256` for proposal Id in Sender as well as Receiver contract.